src/libasr/asr_utils.h

```diff
@@ -1648,6 +1657,11 @@ inline int extract_dimensions_from_ttype(ASR::ttype_t *x,
             n_dims = extract_dimensions_from_ttype(ASR::down_cast<ASR::Const_t>(x)->m_type, m_dims);
             break;
         }
+        case ASR::ttypeType::SymbolicExpression: {
+            n_dims = 0;
+            m_dims = nullptr;
+            break;
+        }
```

Comment:
This can be merged with the below case. For example:
```suggestion
        case ASR::ttypeType::SymbolicExpression:
```